lib/setupHooks/inheritCargoArtifactsHook.sh

```diff
@@ -21,6 +21,10 @@ inheritCargoArtifacts() {
 
     zstd -d "${preparedArtifacts}" --stdout | \
       tar -x -C "${cargoTargetDir}" --strip-components=1
+
+    # Make cached artifacts writable. Some crates' build scripts assume this, such as bzip2-sys
+    # (https://github.com/alexcrichton/bzip2-rs/blob/3032f3790742bffda521e54d14429f459e078eba/bzip2-sys/build.rs#L44)
+    chmod -R u+w "${cargoTargetDir}"
```

Comment:
I wonder if something can be done about `tar` invocation to avoid another pass here over `./target`.

Reply:
Also - why are these files not writable in the first place? `tar` should archive them as they were originally, no?

Reply:
Possibly something about umask. I can check later.

Reply:
[Okay, so tar does preserve permissions, but applies the target directories umask when extracting.](https://stackoverflow.com/a/20414651/577485)

But as far as I understand, the umask is applied only as a bitwise or, i.e. only adding permissions. But I may be wrong about this.

So this would mean that the files must be read-only when compressing already... I can check that when I am back at the PC.

Reply:
```
However, the umask can be ignored by using the -p (--preserve) option, e.g.:
```

Sounds like a good idea to me.

Reply:
Also:

>So unless the user's umask is 000, then the permissions will be updated at that point.

Which implies that the umask can only add permissions, not remove

Reply:
Yes, so I tried replacing the `chmod` command with calling `tar` with `-p`. This makes the compilation fail again. So this must mean that the permissions are wrong in the tarball already. Probably because they are set to read-only by `fs::copy`.

Reply:
So I changed the mode to change already while compressing, and that seems to work as well, and should not introduce any noticable overhead.